mapie/metrics.py

```diff
@@ -307,3 +311,407 @@ def top_label_ece(
         )
     ece /= len(labels)
     return ece
+
+
+def regression_coverage_score_v2(
+    y_true: NDArray,
+    y_intervals: NDArray,
+) -> NDArray:
+    """
+    Effective coverage score obtained by the prediction intervals.
+
+    The effective coverage is obtained by estimating the fraction
+    of true labels that lie within the prediction intervals.
+
+    It is different from ``regression_coverage_score`` because it uses
+    directly the output of ``predict`` method and can compute the
+    coverage for each alpha.
+
+    Parameters
+    ----------
+    y_true: NDArray of shape (n_samples,)
+        True labels.
+    y_intervals: NDArray of shape (n_samples, 2, n_alpha)
+        Lower and upper bound of prediction intervals
+        with different alpha risks.
+
+    Returns
+    -------
+    NDArray of shape (n_alpha,)
+        Effective coverage obtained by the prediction intervals.
+    """
+    coverages = np.mean(
+        np.logical_and(
+            np.less_equal(y_intervals[:, 0, :], y_true),
+            np.greater_equal(y_intervals[:, 1, :], y_true)
+        ),
+        axis=0
+    )
+    return coverages
+
+
+def classification_coverage_score_v2(
+    y_true: NDArray,
+    y_pred_set: NDArray
+) -> NDArray:
+    """
+    Effective coverage score obtained by the prediction sets.
+
+    The effective coverage is obtained by estimating the fraction
+    of true labels that lie within the prediction sets.
+
+    It is different from ``classification_coverage_score`` because it uses
+    directly the output of ``predict`` method and can compute the
+    coverage for each alpha.
+
+    Parameters
+    ----------
+    y_true: NDArray of shape (n_samples,)
+        True labels.
+    y_pred_set: NDArray of shape (n_samples, n_class, n_alpha)
+        Prediction sets given by booleans of labels.
+
+    Returns
+    -------
+    NDArray of shape (n_alpha,)
+        Effective coverage obtained by the prediction sets.
+    """
+    if len(y_true.shape) < len(y_pred_set.shape):
+        y_true = np.expand_dims(y_true, axis=1)
```

Comment:
Isn't it always true ? because according ot your docstring, y_true is of shape (n_samples, ) and y_pred_set (n_samples, n_class, n_alpha)

Reply:
Yes, but in reality the function accepts y_pred_set of shape (n_samples, n_classes), should I change the docstring ?

Reply:
It's important that the user calls the function correctly. I recommend a test that checks the shape of the input.

Reply:
It's fixed, the function now requires y_true of shape (n_samples, n_alpha) and checks the shape of y_pred_set and y_true.